js/pro/poloniex.js

```diff
@@ -0,0 +1,959 @@
+            'event': 'subscribe',
+            'channel': [
+                name,
+            ],
+        };
+        const marketIds = [ ];
+        if (symbols !== undefined) {
+            for (let i = 0; i < symbols.length; i++) {
+                const symbol = symbols[i];
+                marketIds.push (this.marketId (symbol));
+            }
+        } else {
+            marketIds.push ('all');
+        }
+        if (name !== 'balances') {
+            subscribe['symbols'] = marketIds;
+        }
+        const messageHash = name + ':' + marketIds.join (',');
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
+    }
+
+    async watchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOHLCV
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://docs.poloniex.com/#public-channels-market-data-candlesticks
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns [[int]] A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        const channels = this.options['ohlcvChannels'];
+        const channel = this.safeString (channels, timeframe);
+        if (channel === undefined) {
+            throw new BadRequest (this.id + ' watchOHLCV cannot take a timeframe of ' + timeframe);
+        }
+        const ohlcv = await this.subscribe (channel, false, [ symbol ], params);
+        if (this.newUpdates) {
+            limit = ohlcv.getLimit (symbol, limit);
+        }
+        return this.filterBySinceLimit (ohlcv, since, limit, 0, true);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://docs.poloniex.com/#public-channels-market-data-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const name = 'ticker';
+        return await this.subscribe (name, false, [ symbol ], params);
+    }
+
+    async watchTickers (symbols, params = {}) {
```

Comment:
I'd allow to pass an undefined symbols to allow the user to subscribe to all symbols
```suggestion
    async watchTickers (symbols: string[] = undefined, params = {}) {
```